tests/test_dle.py

```diff
@@ -737,3 +737,67 @@ def test_multiple_subnests_v1(self):
         assert trees[1][3].pragmas[0].value == ('omp parallel for collapse(1) '
                                                 'schedule(dynamic,1) '
                                                 'num_threads(nthreads_nested)')
+
+    def test_nested_cache_blocking_structure_subdims(self):
```

Comment:
there's a huge fraction of duplicated code in this test, it's like two tests in one. This is quite ugly.
 
I would rather parametrize over `blocklevels`. You may then use the `blocklevels` value (1 or 2) to check for additional entries in `trees`